### ISSUE TITLE
can i use this tools for appium grid ?

### ISSUE THREAD

**Issue by qileilove** [saikrishna321/AppiumTestDistribution#57]
@saikrishna321 
Hi, buddy.
i want to run the test on different PC, Most of the solution is that use selenium grid and support the parallel processing testing framework. Does  this repo support use the selenium grid to run the test on different node?

**Comment by saikrishna321** [comment]
@qileilove right now no .. its just runs locally ... This FW helps you trigger the test in parallel without much setup. 

when u say different node, is it different machine ?

**Comment by qileilove** [comment]
@saikrishna321 yes, is different machine.  thanks agains. hope this repo will be supported grid in future.
BTW,I'm a thoughtworker too. I'm working In Xi'an office.

**Comment by saikrishna321** [comment]
@qileilove thats awesome... 

I'm still curious to know whats the use-case where u want to run the tests in different machine not in the local-machine ?

**Comment by VinitSandesara** [comment]
Hi Sai, Even i am interested on this, so for our case we have 3 mac machines and we wanna distribute test cases per mac machine connected with 3 iOS devices each. So lets say one mac machine where the code is if i run the project on that machine how the test cases will be distribute on connected mac machines.

Reason for keeping 3 mac machine is not to load all test cases on one mac machine, also to get the performance and speed up this will help.

Thanks
Vinit

**Comment by qileilove** [comment]
@saikrishna321 I'm sorry to reply you so late, I just want to know about how the selenium grid works. may be i can develop the mobile cloud testing framework.

**Comment by qileilove** [comment]
@VinitSandesara  you can use testng to distribute test cases for different mac machine.

**Comment by VinitSandesara** [comment]
@qileilove, How in this Sai's framework  ?

Thanks
Vinit

**Comment by qileilove** [comment]
@VinitSandesara sorry for that.... , this framework  just runs locally, selenium grid + appium+testng will be meet your demand.

**Comment by saikrishna321** [comment]
@qileilove its possible to tweak this FW to run on remote machines, but just thinking how do we trigger the appium server on the remote machine

**Comment by VinitSandesara** [comment]
@qileilove, do you have any framework you can provide us that we can give it a try ?

Thanks
Vinit

**Comment by qileilove** [comment]
@saikrishna321 maybe we can use CI(go_cd,jenkins) to monitor the appium server on remote, if server not working, just ssh command to restart the services.

**Comment by qileilove** [comment]
@VinitSandesara Unfortunately, I didn't find any framework.... but some mobile testing cloud  platform(such as saucelabs testdroid) support what you want, it need to spend some money.

**Comment by akhilbojedla** [comment]
@saikrishna321 @qileilove  Yes Sai we can tweak this framework to distribute tests on different machines. I already designed a framework and its running flawlessly. But only requirement is that we need ssh access (I'm using linux and mac) to the remote machine where devices are connected. My framework users appium+selenium grid+mongodb+testng. And I must thank you for the Extent report help. 👍

**Comment by akhilbojedla** [comment]
@saikrishna321  Currently I'm working on dockerizing the Appium server to make life easier as we can start and stop appium server container if we dockerize it.. Will share the docker image soon.. :)

**Comment by saikrishna321** [comment]
@akhilbojedla  thats a great news 👍  if we really need to run tests on distributed machine then having ssh access should not be a problem. I'm interesting in seeing this dude.. 

i had a discussion with @qileilove yesterday on dockerizing it, but that would work only for android tests not iOS tests. Correct me if i'm wrong ?>

**Comment by akhilbojedla** [comment]
@saikrishna321 Distributing tests on remote machine is nothing but a virtual cloud (If not real). And is there any reason you think dockerizing appium server wouldn't support IOS? Because I do not see a blockage as of now. Please let me know if you see any... 

And ssh is not a problem if we use Linux/Mac machines.. But is a problem if the remote machine is windows(And not a problem if windows is the triggering/Master machine and Mac/Linux machines are slaves). I presume you already knew java secure channel. If not have a look at this [link](http://www.jcraft.com/jsch/) . We can use these libraries to connect to remote machine through ssh programatically. Hope this helps.. 👍

**Comment by saikrishna321** [comment]
@akhilbojedla can we talk about this on gitter 

https://gitter.im/saikrishna321/AppiumTestDistribution?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge

**Comment by qileilove** [comment]
@akhilbojedla I means that the docker images have such Ubuntu and other linux images, but have no os x and windows images, for the ios simulator can only run on os x. so appium server can run all the os.but the script can only run on the virtual machine or real device.

**Comment by qileilove** [comment]
@akhilbojedla  Did you finish your work about framework and docker?

**Comment by akhilbojedla** [comment]
@qileilove I think you are getting my point wrong.  Obviously we cannot run scripts on Windows/Linux machine even if the appium is running in docker container.

My target is to run tests on target machine.  So if the target machine is Mac..  We can run tests on IOS too..

**Comment by deepakarorawins** [comment]
@saikrishna321 @qileilove Did we get this framework to run on grid?

**Comment by saikrishna321** [comment]
@deepakarorawins you are commenting on very old issue. Please raise a new issue on what scenario are you trying to do ?

**Comment by deepakarorawins** [comment]
Ok I'll raise a new issue. Thank you.
